### ISSUE TITLE
[@wordpress/scripts]jest-environment-jsdom@29.7.0 has some deprecated methods

### ISSUE THREAD

**Issue by stein2nd** [WordPress/gutenberg#56543]
### Description

1. npm WARN deprecated abab@2.0.6: Use your platform's native atob() and btoa() methods instead
2. npm WARN deprecated domexception@4.0.0: Use your platform's native DOMException instead

### Step-by-step reproduction instructions

1. Delete the "node_modules" folder and "package-lock.json".
2. Execute "ncu", "ncu -u", "npm install --force" in order.

### Screenshots, screen recording, code snippet

![abab@2 0 6 deduped](https://github.com/WordPress/gutenberg/assets/531628/7eda5ecc-cb07-418b-bbed-d16e988ffbbd)

![jest-environment-jsdom@29 7 0](https://github.com/WordPress/gutenberg/assets/531628/21e1c221-8842-4578-9e1a-fd464a0f0dc2)

## System:
 - OS: macOS 14.1.1
 - CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
 - Memory: 1.26 GB / 32.00 GB
 - Shell: 3.2.57 - /bin/bash
## Binaries:
 - Node: 21.2.0 - ~/.nodebrew/current/bin/node
 - Yarn: 1.22.21 - /usr/local/bin/yarn
 - npm: 10.2.4 - ~/.nodebrew/current/bin/npm
## npmPackages:
 - @webpack-cli/generators: ^3.0.7 => 3.0.7 
 - babel-loader: ^9.1.3 => 9.1.3 
 - clean-webpack-plugin: ^4.0.0 => 4.0.0 
 - copy-webpack-plugin: ^11.0.0 => 11.0.0 
 - css-loader: ^6.8.1 => 6.8.1 
 - css-minimizer-webpack-plugin: ^5.0.1 => 5.0.1 
 - eslint-webpack-plugin: ^4.0.1 => 4.0.1 
 - postcss-loader: ^7.3.3 => 7.3.3 
 - sass-loader: ^13.3.2 => 13.3.2 
 - source-map-loader: ^4.0.1 => 4.0.1 
 - style-loader: ^3.3.3 => 3.3.3 
 - stylelint-webpack-plugin: ^4.1.1 => 4.1.1 
 - terser-webpack-plugin: ^5.3.9 => 5.3.9 
 - ts-loader: ^9.5.1 => 9.5.1 
 - webpack: ^5.89.0 => 5.89.0 
 - webpack-bundle-analyzer: ^4.10.1 => 4.10.1 
 - webpack-cli: ^5.1.4 => 5.1.4 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by dawidurbanski** [comment]
I just looked into this and this is what I found:

```
└─┬ source-map-loader 3.0.2
  └── abab 2.0.6
```

`source-map-loader` removed the `abab` package dependency in version [4.0.2](https://github.com/webpack-contrib/source-map-loader/releases/tag/v4.0.2). Maybe it's worth to give it a shot and update `3.0.2` -> `4.0.2`.

I made [PR](https://github.com/WordPress/gutenberg/pull/58904) just in case it's ok to do this.

```
├─┬ jest-environment-jsdom 29.7.0
│ └─┬ jsdom 20.0.3
│   ├── abab 2.0.6
│   └─┬ data-urls 3.0.2
│     └── abab 2.0.6
```

```
  └─┬ jsdom 20.0.3
    └── domexception 4.0.0
```

This one is more tricky because `@wordpress/scripts` uses the latest version of the `jest-environment-jsdom` package (at the time of writing this comment).

But `jsdom` and its dependency `data-urls` both already removed the `abab` and `domexception` dependency.

Since this has been solved by these I created [an issue in `jest-environment-jsdom` repository](https://github.com/jestjs/jest/issues/14890). If they will fix it, this could be unblocked.

EDIT: I also created [pull request](https://github.com/jestjs/jest/pull/14891) in `jest` repository to bump `jsdom` package.

**Comment by youknowriad** [comment]
Closing as resolved: @wordpress/scripts now depends on jest-environment-jsdom ^30.2.0, which moved past the deprecated jsdom APIs this flagged. Fresh deprecation warnings from the current toolchain would deserve a new issue. Thanks!
